### PR TITLE
Bug #56499: unreadable /etc/localtime should not cause an exception

### PR DESCRIPTION
**Pull request by saper**
Handle exceptions better

### CODE REVIEW

File: mcs/class/corlib/System/TimeZoneInfo.cs [hunk 1]
```diff
@@ -574,11 +574,17 @@ public static TimeZoneInfo FindSystemTimeZoneById (string id)
...
-			using (FileStream stream = File.OpenRead (filepath)) {
+			FileStream stream = null;
+			try {
+				try {
+					stream = File.OpenRead (filepath);	
+				} catch (Exception) {
+					throw new TimeZoneNotFoundException ();
+				}
 				return BuildFromStream (id, stream);
+			} finally {
```
Comment (marek-safar) [inline mcs/class/corlib/System/TimeZoneInfo.cs]: Should be try/finally over BuildFromStream only

File: mcs/class/corlib/System/TimeZoneInfo.cs [hunk 2]
```diff
@@ -574,11 +574,17 @@ public static TimeZoneInfo FindSystemTimeZoneById (string id)
...
-			if (!File.Exists (filepath))
-				throw new TimeZoneNotFoundException ();
-
-			using (FileStream stream = File.OpenRead (filepath)) {
+			FileStream stream = null;
+			try {
+				try {
+					stream = File.OpenRead (filepath);	
+				} catch (Exception) {
+					throw new TimeZoneNotFoundException ();
```
Comment (marek-safar) [inline mcs/class/corlib/System/TimeZoneInfo.cs]: Please include caught exception in TimeZoneNotFoundException

**Comment by saper** [comment]
Thank you for fixing this @akoeplinger !

**Comment by EvilBeaver** [comment]
Guys, when this fix can be retrieved from mono apt repository? Which version it should be?
Or, maybe you can tell me version to which I should downgrade?

**Comment by akoeplinger** [comment]
@EvilBeaver it will be in mono 5.2 (available in the alpha channel).

**Pull request merged**